march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -323,30 +316,12 @@ void MarchHardwareInterface::initiateIMC()
   for (const std::string& joint_name : joint_names_)
   {
     Joint joint = marchRobot.getJoint(joint_name);
-
-    if (LOWER_BOUNDARY_ANGLE_IU <= joint.getAngleIU() && joint.getAngleIU() <= UPPER_BOUNDARY_ANGLE_IU)
-    {
-      ROS_WARN("Before reset joint: [%s] has angle-value of: %i. Which is within boundary of lower: %i and upper: %i",
-               joint_name.c_str(), joint.getAngleIU(), LOWER_BOUNDARY_ANGLE_IU, UPPER_BOUNDARY_ANGLE_IU);
-    }
-
     joint.resetIMotionCube();
   }
 
   ROS_INFO("Restarting EtherCAT");
   marchRobot.stopEtherCAT();
   marchRobot.startEtherCAT();
-
-  for (const std::string& joint_name : joint_names_)
-  {
-    Joint joint = marchRobot.getJoint(joint_name);
-
-    if (LOWER_BOUNDARY_ANGLE_IU <= joint.getAngleIU() && joint.getAngleIU() <= UPPER_BOUNDARY_ANGLE_IU)
-    {
-      ROS_WARN("After reset joint: [%s] has angle-value of: %i. Which is within boundary of lower: %i and upper: %i",
-               joint_name.c_str(), joint.getAngleIU(), LOWER_BOUNDARY_ANGLE_IU, UPPER_BOUNDARY_ANGLE_IU);
-    }
-  }
```

Comment:
If after the reset the angles are still a non valid value shouldn't there be an error? I know this was not implemented before but maybe we should?

Reply:
We have removed these lines, because you'd be reading out encoder values before the iMC is in SWITCHED_ON state.